fud/fud/stages/verilator/json_to_dat.py

```diff
@@ -133,14 +133,14 @@ def convert2dat(output_dir, data, extension, round: bool):
         numeric_type = format["numeric_type"]
         is_signed = format["is_signed"]
 
-        if numeric_type not in {"bitnum", "fixed_point"}:
-            raise InvalidNumericType('Fud only supports "fixed_point" and "bitnum".')
+        if numeric_type not in {"bitnum", "fixed_point", "float_point"}:
+            raise InvalidNumericType('Fud only supports "fixed_point", "bitnum", and "float_point".')
 
         is_fp = numeric_type == "fixed_point"
         if is_fp:
             width, int_width = parse_fp_widths(format)
         else:
-            # `Bitnum`s only have a bit width.
+            # `Bitnum`s and `FloatPoint`s only have a bit width
```

Comment:
I'm a little confused about how `float_point` is handled here… it seems like we are passing them through unmodified? So the user has to specify a decimal integer that corresponds to the appropriate float-point bit pattern?

I guess this is an OK way to get started, but in a separate PR, maybe we would like to make floating point work more like fixed point… as in, the JSON file contains actual floating-point numbers, which we then emit as bits.

Reply:
Currently, floating point number representation is still handled very "primitively" using binary number by the IEEE 754 standard, since it's the easiest way.
> So the user has to specify a decimal integer that corresponds to the appropriate float-point bit pattern?

So the user has to specify a long bit string, which can be a pain for them...
I was thinking to support decimal representation so user can write `4.2` in the json file in the future, if that's what you mean by
> as in, the JSON file contains actual floating-point numbers, which we then emit as bits.